Readme.md

```diff
@@ -1,65 +1,118 @@
+
+![Upload Control](./images/uploadControl.gif "UploadControl")
+
+## Overview:
+
+Follow the steps below to allow users to upload multiple files and display links to uploaded files on a web page:
+
+1. Add the [Upload Control extension](https://docs.devexpress.com/AspNetMvc/DevExpress.Web.Mvc.UploadControlExtension?p=netframework) to a view. To do this, you can use the [Insert Extension Wizard](https://docs.devexpress.com/AspNetMvc/14551/whats-installed/insert-devexpress-mvc-extension-wizard) or write the extension code manually.
 
-### The MVC [UploadControl](https://documentation.devexpress.com/AspNet/9002/ASP-NET-MVC-Extensions/File-Management/File-Upload/Overview/Overview-UploadControl) configuration here allows you to upload multiple files at once. It also supports drag and drop and upload progress indication. Links to the uploaded files will be displayed in a special container.
+2. In the *Global.asax* or *RouteConfig* file, add the **.ashx** file extension to route exceptions:
 
-## Steps to implement:
-1. Add the UploadControl extension to View. The easiest way is to use the [*Insert Extension Wizard*](https://documentation.devexpress.com/AspNet/14551/What-s-Installed/Visual-Studio-Integration/Insert-MVC-Extension-Wizard/Insert-DevExpress-MVC-Extension-Wizard), but you can write code manually as well.
-2. Wrap the control in the "form" HTML tag. In this example, the [*Html.BeginForm*](https://docs.microsoft.com/en-us/dotnet/api/system.web.mvc.html.formextensions.beginform?view=aspnet-mvc-5.2) method is used to create a form.
-3. Specify the Action method that handles file uploading. E.g.:
-```cs
-@using (Html.BeginForm("UploadControlUpload", "Home")){
+    ```cs
+    routes.IgnoreRoute("{resource}.axd/{*pathInfo}");
+    routes.IgnoreRoute("{resource}.ashx/{*pathInfo}");
+    ```
+
+3. Place the [Round Panel extension](https://docs.devexpress.com/AspNetMvc/DevExpress.Web.Mvc.RoundPanelExtension?p=netframework) below the Upload Control:
+
+    ```razor
+    @Html.DevExpress().RoundPanel(settings => {
+        settings.Name = "RoundPanel";
+        settings.HeaderText = "Uploaded Files";
+        settings.SetContent(() => { ViewContext.Writer.Write("<div id='fileContainer'/>"); });
+    }).GetHtml()
+    ```
+
+4. Configure settings of the Upload Control extension. Set the extension's [UploadMode](https://docs.devexpress.com/AspNetMvc/DevExpress.Web.Mvc.UploadControlSettings.UploadMode?p=netframework) property to [Advanced](https://docs.devexpress.com/AspNet/DevExpress.Web.UploadControlUploadMode) to enable advanced upload mode. To allow users to upload multiple files, set the [EnableMultiSelect](https://docs.devexpress.com/AspNet/DevExpress.Web.UploadAdvancedModeSettings.EnableMultiSelect?p=netframework) property to `true`. The [EnableDragAndDrop](https://docs.devexpress.com/AspNet/DevExpress.Web.UploadAdvancedModeSettings.EnableDragAndDrop) property specifies whether to enable drag-and-drop support.
```

Comment:
I've always been a consistent adversary of the pattern "whether to do something". Is there any specific reason for using it?

What does this property specify? It specifies whether support for drag-and-drop operation is active. Nothing more. It just declares if the said functionality is disabled or enabled, it does not allow your component to decide on its own.

We should be extremely to-the-point in everything we write. Phrases like "whether to enable" leave too much room for interpretation, and use more words as well.

Whether == if, if is shorter. However, I don't have any objections against the word "whether" itself, just this pattern "whether to do something".

Can we simplify this sentence further? We can. Can we deduce "enabled drag-and-drop operations" from "enabled drag-and-drop support"? Yes, we can. While the final result may not be much shorter in terms of the overall character count, it has more ties with operations made by users directly.

```suggestion
4. Configure settings of the Upload Control extension. Set the extension's [UploadMode](https://docs.devexpress.com/AspNetMvc/DevExpress.Web.Mvc.UploadControlSettings.UploadMode?p=netframework) property to [Advanced](https://docs.devexpress.com/AspNet/DevExpress.Web.UploadControlUploadMode) to enable advanced upload mode. To allow users to upload multiple files, set the [EnableMultiSelect](https://docs.devexpress.com/AspNet/DevExpress.Web.UploadAdvancedModeSettings.EnableMultiSelect?p=netframework) property to `true`. The [EnableDragAndDrop](https://docs.devexpress.com/AspNet/DevExpress.Web.UploadAdvancedModeSettings.EnableDragAndDrop) property specifies if drag-and-drop operations are enabled.
```